sql/partitioning/partition_base.cc

```diff
@@ -4046,6 +4049,31 @@ bool Partition_base::commit_inplace_alter_table(
   if (ha_alter_info->alter_info->flags == Alter_info::ALTER_PARTITION)
     DBUG_RETURN(false);
 
+  std::string full_table_path = std::string("./") + altered_table->s->db.str +
+                                std::string("/") +
+                                altered_table->s->table_name.str;
+  if (ha_alter_info->alter_info->flags & Alter_info::ALTER_DROP_PARTITION) {
+    handler **file = m_file;
+    partition_info *part_info = ha_alter_info->modified_part_info;
+    List_iterator_fast<partition_element> part_it(part_info->partitions);
+    partition_element *part_elem;
+    while ((part_elem = part_it++) != nullptr) {
+      partition_state state = part_elem->part_state;
+      switch (state) {
+        case PART_TO_BE_DROPPED:
+          char name[FN_REFLEN];
+          // TODO(luqun): add subpartition support
+          create_partition_name(name, full_table_path.c_str(),
+                                part_elem->partition_name, false);
+          (*file)->ha_external_lock(get_thd(), F_UNLCK);
+          (*file)->ha_delete_table(name, old_table_def);
```

Comment:
Please check if concurrent SELECTs can be done.  Does the delete_table() implementation in myrocks gurantee to wait until pending IO is completed?

Reply:
myrocks is "append only" database and there is no background threads to service IO request. AFAIK, it doesn't need to wait for pending IO.

Reply:
> Please check if concurrent SELECTs can be done. 

there are two small time windows drop partition will acquire/hold exclusive table lock, During these two time windows, SELECT will wait for acquire table lock if it doesn't hold table lock yet.